tests/integration/test_packagewagl.py

```diff
@@ -61,42 +61,42 @@ def test_whole_wagl_package(
     assert_file_structure(
         expected_folder,
         {
-            "ga_ls8c_ard_3-0-1_092084_2016-06-28_final.odc-metadata.yaml": "",
-            "ga_ls8c_ard_3-0-1_092084_2016-06-28_final.proc-info.yaml": "",
-            "ga_ls8c_ard_3-0-1_092084_2016-06-28_final.sha1": "",
-            "ga_ls8c_nbar_3-0-1_092084_2016-06-28_final_band01.tif": "",
-            "ga_ls8c_nbar_3-0-1_092084_2016-06-28_final_band02.tif": "",
-            "ga_ls8c_nbar_3-0-1_092084_2016-06-28_final_band03.tif": "",
-            "ga_ls8c_nbar_3-0-1_092084_2016-06-28_final_band04.tif": "",
-            "ga_ls8c_nbar_3-0-1_092084_2016-06-28_final_band05.tif": "",
-            "ga_ls8c_nbar_3-0-1_092084_2016-06-28_final_band06.tif": "",
-            "ga_ls8c_nbar_3-0-1_092084_2016-06-28_final_band07.tif": "",
-            "ga_ls8c_nbar_3-0-1_092084_2016-06-28_final_band08.tif": "",
-            "ga_ls8c_nbar_3-0-1_092084_2016-06-28_final_thumbnail.jpg": "",
-            "ga_ls8c_nbart_3-0-1_092084_2016-06-28_final_band01.tif": "",
-            "ga_ls8c_nbart_3-0-1_092084_2016-06-28_final_band02.tif": "",
-            "ga_ls8c_nbart_3-0-1_092084_2016-06-28_final_band03.tif": "",
-            "ga_ls8c_nbart_3-0-1_092084_2016-06-28_final_band04.tif": "",
-            "ga_ls8c_nbart_3-0-1_092084_2016-06-28_final_band05.tif": "",
-            "ga_ls8c_nbart_3-0-1_092084_2016-06-28_final_band06.tif": "",
-            "ga_ls8c_nbart_3-0-1_092084_2016-06-28_final_band07.tif": "",
-            "ga_ls8c_nbart_3-0-1_092084_2016-06-28_final_band08.tif": "",
-            "ga_ls8c_nbart_3-0-1_092084_2016-06-28_final_thumbnail.jpg": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_azimuthal-exiting.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_azimuthal-incident.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_combined-terrain-shadow.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_exiting-angle.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_fmask.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_incident-angle.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_nbar-contiguity.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_nbart-contiguity.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_relative-azimuth.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_relative-slope.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_satellite-azimuth.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_satellite-view.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_solar-azimuth.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_solar-zenith.tif": "",
-            "ga_ls8c_oa_3-0-1_092084_2016-06-28_final_time-delta.tif": "",
+            "ga_ls8c_ard_3-1-20190711_092084_2016-06-28_final.odc-metadata.yaml": "",
```

Comment:
I'm ok with a date being included as part of the version number `3-1-20190711`, however, that looks like a `yyyymmdd` not a `yyyyddd` for day of year.  Do we need that level though.  Or simply `3.1.0`, and if it gets reprocessed it is `3.1.1`, and if we do an algorithm update, it'll be `3.2.0`.

Reply:
The tough thing about a numerical dataset version is how to track the reprocesses.

Currently, the processing software works on its own with no knowledge of the outside world.

In order to know that this is exactly the _third_ version of a dataset, it would need to be able to look that up somewhere, which introduces dependencies. A date-based version, like USGS already does with L1, is a much simpler and more reliable solution.

I don't mind YYDOY vs YYMMDD, but I find the latter much easier to understand at a glance ("this is from november") -- no need for conversions, and it's only one character more.

Reply:
The latter is definitely more understandable, and I'd favour that version.  The DOY might be what USGS use for acquistion date, not processing date.

Reply:
Go with `YYYYMMDD` as you've already done.